### PR TITLE
KAFKA-10044 Deprecate ConsumerConfig#addDeserializerToConfig and Prod…

### PR DESCRIPTION
**Pull request by chia7712**
issue: https://issues.apache.org/jira/browse/KAFKA-10044

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java [hunk 1]
```diff
@@ -579,9 +579,19 @@ private void maybeOverrideClientId(Map<String, Object> configs) {
...
+     * @deprecated Since 2.7.0. This will be removed in a future major release.
+     */
+    @Deprecated
     public static Map<String, Object> addDeserializerToConfig(Map<String, Object> configs,
                                                               Deserializer<?> keyDeserializer,
                                                               Deserializer<?> valueDeserializer) {
+        return appendDeserializerToConfig(configs, keyDeserializer, valueDeserializer);
+    }
+
+    static Map<String, Object> appendDeserializerToConfig(Map<String, Object> configs,
```
Comment (abbccdda) [inline clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java:592]: align parameters

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java [hunk 2]
```diff
@@ -492,8 +492,17 @@ private static String parseAcks(String acksString) {
...
+    /**
+     * @deprecated Since 2.7.0. This will be removed in a future major release.
+     */
+    @Deprecated
     public static Map<String, Object> addSerializerToConfig(Map<String, Object> configs,
                                                             Serializer<?> keySerializer, Serializer<?> valueSerializer) {
+        return appendSerializerToConfig(configs, keySerializer, valueSerializer);
+    }
+
+    static Map<String, Object> appendSerializerToConfig(Map<String, Object> configs,
```
Comment (abbccdda) [inline clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java:504]: Same here, one parameter per line and align

File: clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerConfigTest.java [hunk 3]
```diff
@@ -100,24 +101,24 @@ public void testDeserializerToMapConfig() {
         Map<String, Object> configs = new HashMap<>();
         configs.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializerClass);
         configs.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializerClass);
-        Map<String, Object> newConfigs = ConsumerConfig.addDeserializerToConfig(configs, null, null);
+        Map<String, Object> newConfigs = ConsumerConfig.appendDeserializerToConfig(configs, null, null);
```
Comment (abbccdda) [inline clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerConfigTest.java:104]: nit: change the test name to `testAppendDeserializerToConfig`

File: clients/src/test/java/org/apache/kafka/clients/producer/ProducerConfigTest.java [hunk 4]
```diff
@@ -68,24 +69,24 @@ public void testSerializerToMapConfig() {
         Map<String, Object> configs = new HashMap<>();
         configs.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, keySerializerClass);
         configs.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, valueSerializerClass);
-        Map<String, Object> newConfigs = ProducerConfig.addSerializerToConfig(configs, null, null);
+        Map<String, Object> newConfigs = ProducerConfig.appendSerializerToConfig(configs, null, null);
```
Comment (abbccdda) [inline clients/src/test/java/org/apache/kafka/clients/producer/ProducerConfigTest.java:72]: Similar here, change test name to match function.

**Comment by chia7712** [comment]
@abbccdda thanks for reviews! I have addressed your comment. pleas take a look :)

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java [hunk 5]
```diff
@@ -492,8 +492,19 @@ private static String parseAcks(String acksString) {
...
+     * @deprecated Since 2.7.0. This will be removed in a future major release.
+     */
+    @Deprecated
     public static Map<String, Object> addSerializerToConfig(Map<String, Object> configs,
                                                             Serializer<?> keySerializer, Serializer<?> valueSerializer) {
+        return appendSerializerToConfig(configs, keySerializer, valueSerializer);
+    }
+
+    static Map<String, Object> appendSerializerToConfig(
+            Map<String, Object> configs,
```
Comment (abbccdda) [inline clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java]: nit: I don't feel strong about the style here, but maybe we should consider align with other functions which has the first parameter on the same line with function name.

**Review by abbccdda — APPROVED** [review]
LGTM

**Pull request merged**